### DISCUSSION TITLE
Audio output toggle

### DISCUSSION THREAD

**Discussion by Lippsson** [MiczFlor/RPi-Jukebox-RFID#2227] [Q&A] [answered]
Hi community!

I have a question as non developer: How i have to modify the components/bluetooth-sink-switch/bluetooth-sink-switch.py to toogle between Output 1 and Output 2 even if i have no BT-Headset connected?

I have successfully the hifiberry on output 1 and the Raspberry headphone 3.5mm output on output 2
I also be able to toggle from in shell with the `mpc enable only 1` or `mpc enable only 2` command between the 2 outputs

![image](https://github.com/MiczFlor/RPi-Jukebox-RFID/assets/48052340/fdf75e65-8c71-487e-8abf-833fb49ed225)
in the picture above i tested to force `isBTConnected = True` but then i can´t switch back to hifiberry ;-)

**Comment by AlvinSchiller** [comment]
Hi, 
setting `isBTConnected = True` should do the trick as far as i unterstand the code. 
https://github.com/MiczFlor/RPi-Jukebox-RFID/blob/01734deb7c872c2ef241eb6b5c1990231aeee062/components/bluetooth-sink-switch/bt-sink-switch.py#L137

Do you maybe have debug logs?

**Reply by Lippsson** [reply]
Hello Alvin! Thank you for your answer. Ok i changed the line 137 to:  `isBtConnected = True #re.search(b"Connected:\s+yes", isBtConnected_console.stdout) `
This is my debug output:
 ```
pi@barn:~/RPi-Jukebox-RFID/components/bluetooth-sink-switch $ sudo python3 bt-sink-switch.py toggle debug
01.02.2024 22:00:13 - bt-sink-switch.py - DEBUG: Reading config file: '/home/pi/RPi-Jukebox-RFID/components/bluetooth-sink-switch/../../settings/gpio_settings.ini'
01.02.2024 22:00:13 - bt-sink-switch.py - DEBUG: No section BluetoothToggleLed found. Defaulting to led_pin = None
01.02.2024 22:00:13 - bt-sink-switch.py - DEBUG: Using LED pin = None
01.02.2024 22:00:13 - bt-sink-switch.py - DEBUG: b'Output 1 (HiFiBerry DAC+ Lite) is enabled\nOutput 2 (Headphone) is disabled\n'
01.02.2024 22:00:13 - bt-sink-switch.py - DEBUG: b'Missing device address argument\n'
Switched audio sink to "Output 2"
01.02.2024 22:00:13 - bt-sink-switch.py - DEBUG: b'Output 1 (HiFiBerry DAC+ Lite) is enabled\nOutput 2 (Headphone) is enabled\nOutput 1 (HiFiBerry DAC+ Lite) is disabled\nOutput 2 (Headphone) is enabled\n'
01.02.2024 22:00:13 - bt-sink-switch.py - DEBUG: bt_check_mpc_err()
01.02.2024 22:00:13 - bt-sink-switch.py - DEBUG: b'Parcels - Comedown - from Hansa Studios, Berlin\n[playing] #4/18   0:12/3:25 (5%)\nvolume: 43%   repeat: off   random: off   single: off   consume: off\n'
pi@barn:~/RPi-Jukebox-RFID/components/bluetooth-sink-switch $ sudo python3 bt-sink-switch.py toggle debug
01.02.2024 22:00:38 - bt-sink-switch.py - DEBUG: Reading config file: '/home/pi/RPi-Jukebox-RFID/components/bluetooth-sink-switch/../../settings/gpio_settings.ini'
01.02.2024 22:00:38 - bt-sink-switch.py - DEBUG: No section BluetoothToggleLed found. Defaulting to led_pin = None
01.02.2024 22:00:38 - bt-sink-switch.py - DEBUG: Using LED pin = None
01.02.2024 22:00:38 - bt-sink-switch.py - DEBUG: b'Output 1 (HiFiBerry DAC+ Lite) is disabled\nOutput 2 (Headphone) is enabled\n'
01.02.2024 22:00:38 - bt-sink-switch.py - DEBUG: b'Missing device address argument\n'
Switched audio sink to "Output 1"
01.02.2024 22:00:38 - bt-sink-switch.py - DEBUG: b'Output 1 (HiFiBerry DAC+ Lite) is enabled\nOutput 2 (Headphone) is disabled\n'
01.02.2024 22:00:38 - bt-sink-switch.py - DEBUG: bt_check_mpc_err()
01.02.2024 22:00:38 - bt-sink-switch.py - DEBUG: b'Parcels - Comedown - from Hansa Studios, Berlin\n[playing] #4/18   0:38/3:25 (18%)\nvolume:  4%   repeat: off   random: off   single: off   consume: off\n'
pi@barn:~/RPi-Jukebox-RFID/components/bluetooth-sink-switch $

```
It works to switch the output with the toogle debug option but not over the browser: 
![image](https://github.com/MiczFlor/RPi-Jukebox-RFID/assets/48052340/b4ad45d2-ca0e-4005-8b15-eeb66abd0fc2)

**Comment by Lippsson** [comment, marked as answer]
I found the answer with help av ChatGPT. Not easy, but it worked..Haha. I had to `chmod +x /home/pi/RPi-Jukebox-RFID/components/bluetooth-sink-switch/bt-sink-switch.py ` (Maybe cause i changed orginal filename and created a new one with the "bt-sink-switch.py" name)

the changes in bluetooth-sink-switch.py i (ChatGPT) did: 

https://github.com/Lippsson/RPi-Jukebox-RFID/blob/Lippsson-patch-1/components/bluetooth-sink-switch/bt-sink-switch.py

**Reply by AlvinSchiller** [reply]
Glad it works now 👍 

if you created a new file then it has no execution rights by default. ^^ The install script handles this normally.

**Reply by Lippsson** [reply]
yes, thanks for your help
